controllers/spacerequest/spacerequest_controller_test.go

```diff
@@ -25,43 +27,733 @@ import (
+
+			// then
+			require.NoError(t, err)
+			// spacerequest exists with expected cluster roles and finalizer
+			spacerequesttest.AssertThatSpaceRequest(t, srNamespace.Name, sr.GetName(), member1.Client).
+				HasSpecTargetClusterRoles(srClusterRoles).
+				HasConditions(spacetest.Ready()).               // condition is reflected from space status
+				HasStatusTargetClusterURL(member1.APIEndpoint). // has new target cluster url
+				HasFinalizer()
+			// a subspace is created with the tierName and cluster roles from the spacerequest
+			spacetest.AssertThatSpace(t, test.HostOperatorNs, "jane-subs", hostClient).
+				HasSpecTargetClusterRoles(srClusterRoles).
+				HasConditions(spacetest.Ready()).
+				HasTier(sr.Spec.TierName).
+				HasParentSpace("jane") // the parent space is set as expected
+		})
+
+		t.Run("spacerequest created on member2", func(t *testing.T) {
+			// when
+			member1 := NewMemberClusterWithClient(test.NewFakeClient(t), "member-1", corev1.ConditionTrue)
+			member2 := NewMemberClusterWithClient(test.NewFakeClient(t, sr, srNamespace), "member-2", corev1.ConditionTrue) // space request and namespace are on member2
+			hostClient := test.NewFakeClient(t, appstudioTier, parentSpace)
+			ctrl := newReconciler(hostClient, member1, member2)
+			_, err := ctrl.Reconcile(context.TODO(), requestFor(sr))
+
+			// then
+			require.NoError(t, err)
+			// spacerequest exists with expected cluster roles and finalizer
+			spacerequesttest.AssertThatSpaceRequest(t, srNamespace.Name, sr.GetName(), member2.Client).
+				HasSpecTargetClusterRoles(srClusterRoles).
+				HasConditions(spacetest.Provisioning()).
+				HasFinalizer()
+			spacetest.AssertThatSpaces(t, hostClient).ContainsSubSpace(sr, parentSpace)
+		})
 	})
 
 	t.Run("failure", func(t *testing.T) {
-		// TODO implement error cases
+		// given
+		sr := spacerequesttest.NewSpaceRequest("jane",
+			srNamespace.GetName(),
+			spacerequesttest.WithTierName("appstudio"),
+			spacerequesttest.WithTargetClusterRoles(srClusterRoles))
+
+		t.Run("unable to find SpaceRequest", func(t *testing.T) {
+			member1 := NewMemberClusterWithClient(test.NewFakeClient(t), "member-1", corev1.ConditionTrue)
+			hostClient := test.NewFakeClient(t, appstudioTier)
+			ctrl := newReconciler(hostClient, member1)
+
+			// when
+			_, err := ctrl.Reconcile(context.TODO(), requestFor(nil))
+
+			// then
+			// space request should not be there
+			// but it should not return an error by design
+			require.NoError(t, err)
+		})
+		t.Run("unable to get SpaceRequest", func(t *testing.T) {
+			member1Client := test.NewFakeClient(t)
+			member1Client.MockGet = mockGetSpaceRequestFail(member1Client)
+			member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+			hostClient := test.NewFakeClient(t, appstudioTier)
+			ctrl := newReconciler(hostClient, member1)
+
+			// when
+			_, err := ctrl.Reconcile(context.TODO(), requestFor(nil))
```

Comment:
```suggestion
			_, err := ctrl.Reconcile(context.TODO(), requestFor(sr))
```

Reply:
fixed in https://github.com/codeready-toolchain/host-operator/pull/770/commits/0d7e258c62546b6a3de76e1decd116a7c9bb3f2c

Thanks!